### ISSUE TITLE
[bug] REST getutxos is not returning the same data as RPC

### ISSUE THREAD

**Issue by patricklodder** [dogecoin/dogecoin#2853]
While reworking the documentation for the REST interface that I spotted to be not ported with #2770, I found that calling the endpoint `/rest/getutxos` does not return the same results as calling the `gettxout` RPC endpoint. In fact, it looks like there is a bug with it.

# Bug Report

**Current behavior:**

when querying the REST server for a non-mempool output, results are different than when using `gettxout`

```console
$ dogecoin-cli gettxout 7694b4527f00024759a8f614deb0824072c5badcdadfd316f7f425d5eb8199ff 0
{
  "bestblock": "6e8da0b9bf517a428b3fa3e072cf8d723caf0fe810c691f98eda3847b896fde9",
  "confirmations": 3754209,
  "value": 4946744.34700000,
  "scriptPubKey": {
    "asm": "OP_DUP OP_HASH160 5f0d49feb9842f4ef958d1a2f69a776ce17f2808 OP_EQUALVERIFY OP_CHECKSIG",
    "hex": "76a9145f0d49feb9842f4ef958d1a2f69a776ce17f280888ac",
    "reqSigs": 1,
    "type": "pubkeyhash",
    "addresses": [
      "DDogepartyxxxxxxxxxxxxxxxxxxw1dfzr"
    ]
  },
  "version": 1,
  "coinbase": false
}
```

versus

```console
$ curl -s localhost:22555/rest/getutxos/7694b4527f00024759a8f614deb0824072c5badcdadfd316f7f425d5eb8199ff-0.json |jq
{
  "chainHeight": 4108629,
  "chaintipHash": "a47774a7df49d4949c5f9d0c9adb39f498de1bd37445c66847d053f2dffca82b",
  "bitmap": "0",
  "utxos": []
}
```

**Expected behavior**

The REST endpoint returns the same utxo as the RPC method.

**Dogecoin Core version:** v1.14.5 (dogecoind, linux)
**Configuration:** txindex and high mempool expiry (2 weeks)

**Comment by patricklodder** [comment]
@victorsk2019 Yesssssss! Thank you! Want to do the PR?

**Comment by chromatic** [comment]
@victorsk2019 I don't mind writing the PR if you don't want to; I think Patrick wants someone else to do it so he can review it. I'll give you full credit for it if I write it though.

**Comment by patricklodder** [comment]
I have it already. Just waiting on #2860 merge

**Comment by patricklodder** [comment]
Fixed with #2909
